core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorBodyDirective.js

```diff
@@ -38,10 +42,39 @@ oppia.directive('simpleEditorBody', [function() {
         $scope.canAddNewQuestion = SimpleEditorManagerService.canAddNewQuestion;
         $scope.addState = SimpleEditorManagerService.addState;
         $scope.addNewQuestion = SimpleEditorManagerService.addNewQuestion;
-        $scope.canTryToPublish = SimpleEditorManagerService.canTryToPublish;
+        $scope.canTryToFinishExploration =
+          SimpleEditorManagerService.canTryToFinishExploration;
 
-        $scope.tryToPublishExploration = function() {
-          // TODO(sll): Implement this.
+        $scope.isExplorationFinishable = function() {
+          if (explorationRightsService.isPrivate()) {
+            if (!explorationWarningsService.countWarnings()) {
+              return true;
+            }
+          } else if (explorationSaveService.isExplorationSaveable()) {
+            return true;
+          }
+
+          return false;
+        };
+
+        $scope.startFinishExploration = function() {
+          // If exploration is not yet published
+          // and doesn't have unsaved changes,
+          // we can just open publishing modal straight away.
+          if (explorationRightsService.isPrivate() &&
+              !explorationSaveService.isExplorationSaveable()) {
+            explorationSaveService.showPublishExplorationModal();
+          } else {
+            explorationSaveService.saveChanges()
+              .then(function(didSaveExploration) {
```

Comment:
(Optional) Perhaps saveSucceeded? (I think "if the save succeeded" sounds more natural than "if did save exploration".)